cache/public/navigation.go

```diff
@@ -0,0 +1,40 @@
+package public
+
+import (
+	"context"
+
+	"github.com/ONSdigital/dp-frontend-search-controller/config"
+	topicModel "github.com/ONSdigital/dp-topic-api/models"
+	topicCli "github.com/ONSdigital/dp-topic-api/sdk"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+func UpdateNavigationData(ctx context.Context, cfg *config.Config, lang string, topicClient topicCli.Clienter) func() *topicModel.Navigation {
+	if cfg.EnableNewNavBar == false {
+		return func() *topicModel.Navigation {
+			return &topicModel.Navigation{}
+		}
+	}
+	return func() *topicModel.Navigation {
```

Comment:
Why are we returning a `func() *topicModel.Navigation`. Can we just not return `*topicModel.Navigation`?

Reply:
This is what ultimately gets handed to the dp-cache update function, so I think it needs to be structured this way. Perhaps there's a more efficient way?

Reply:
This makes sense. Also, looked at where this function is used and it all links up